watertap/unit_models/nanofiltration0D.py

```diff
@@ -0,0 +1,654 @@
+        "multivalent_recovery": 1e2,
+        "solvent_recovery": 10,
+        "solute_recovery": 10,
+    }
+
+    def variable_scaling_routine(
+        self, model, overwrite: bool = False, submodel_scalers: dict = None
+    ):
+        # Scale inlet state
+        self.call_submodel_scaler_method(
+            submodel=model.properties_in,
+            method="variable_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+
+        # Propagate scaling from inlet state
+        self.propagate_state_scaling(
+            target_state=model.properties_retentate,
+            source_state=model.properties_in,
+            overwrite=overwrite,
+        )
+        self.propagate_state_scaling(
+            target_state=model.properties_permeate,
+            source_state=model.properties_in,
+            overwrite=overwrite,
+        )
+
+        # Scale remaining states
+        self.call_submodel_scaler_method(
+            submodel=model.properties_retentate,
+            method="variable_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+        self.call_submodel_scaler_method(
+            submodel=model.properties_permeate,
+            method="variable_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+
+        # Scale remaining variables
+        # Variables with default scaling factors
+        for n in self.DEFAULT_SCALING_FACTORS.keys():
+            c = model.find_component(n)
+            for v in c.values():
+                self.scale_variable_by_default(v, overwrite=overwrite)
+
+    def constraint_scaling_routine(
+        self, model, overwrite: bool = False, submodel_scalers: dict = None
+    ):
+        # Call scaling methods for sub-models
+        self.call_submodel_scaler_method(
+            submodel=model.properties_in,
+            method="constraint_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+        self.call_submodel_scaler_method(
+            submodel=model.properties_retentate,
+            method="constraint_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+        self.call_submodel_scaler_method(
+            submodel=model.properties_permeate,
+            method="constraint_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+
+        # Scale remaining constraints
+        for c in model.component_data_objects(Constraint, descend_into=False):
+            self.scale_constraint_by_nominal_value(
+                c,
+                scheme=ConstraintScalingScheme.inverseMaximum,
+                overwrite=overwrite,
+            )
+
+
+@declare_process_block_class("Nanofiltration0D")
+class Nanofiltration0DData(UnitModelBlockData):
```

Comment:
My high-level question before diving in more deeply is-- why do we need another NF model class? Could we instead just modify the existing NF_ZO model and potentially rename if needed? Or could we delete NF_ZO and replace with this? 

My concern is that all of the different NF files that we have now can already be confusing, and adding another separate NF file will add to that confusion. So it'd be best if we could consolidate.

Reply:
After the fact, I am asking myself if it would have been better to merge them. In part I did this to ensure backward compatibility as this model makes a few additional assumptions (both to simplify and to enforce electroneutrality).